hibernate-core/src/main/java/org/hibernate/sql/ast/tree/expression/ColumnReference.java

```diff
@@ -25,32 +25,43 @@
  * Models a reference to a Column in a SQL AST
  *
  * @author Steve Ebersole
+ * @author Nathan Xu
  */
 public class ColumnReference implements Expression, Assignable {
 	private final String qualifier;
 	private final String columnExpression;
+	private final boolean isColumnExpressionFormula;
```

Comment:
AFAICS you are not using this so IMO you can remove this. It's enough to check in the constructor but there is no need for the state later

Reply:
Yeah, currently it is not used publicly, but I do think it is a basic feature of this class and we are supposed to expose it for later use.